File: ailist/src/main/java/me/kosik/interwalled/ailist/model/AIListConfiguration.java
package me.kosik.interwalled.ailist.model;


/**
 *
 * @param maximumComponentsCount
 * @param intervalsCountToCheckLookahead
 * @param intervalsCountToTriggerExtraction
 * @param maximumComponentSize
 * @param isInputDataSorted
 */
public record AIListConfiguration(
        int maximumComponentsCount,
        int intervalsCountToCheckLookahead,
        int intervalsCountToTriggerExtraction,
        int maximumComponentSize,
        boolean checkLookbehindCoverage,
        boolean isInputDataSorted
) {

    final public static AIListConfiguration DEFAULT =
            new AIListConfiguration(32, 24, 16, 64, false, false);
}

File: ailist/src/main/scala/me/kosik/interwalled/ailist/model/AIListConfiguration.scala
package me.kosik.interwalled.ailist.model

case class AIListConfiguration(
  maximumComponentsCount: Int,
  intervalsCountToCheckLookahead: Int,
  intervalsCountToTriggerExtraction: Int,
  maximumComponentSize: Int,
  checkLookbehindCoverage: Boolean,
  isInputDataSorted: Boolean
)

object AIListConfiguration {
  def apply: AIListConfiguration = AIListConfiguration(
    maximumComponentsCount = 32,
    intervalsCountToCheckLookahead = 24,
    intervalsCountToTriggerExtraction = 16,
    maximumComponentSize = 64,
    checkLookbehindCoverage = false,
    isInputDataSorted = false
  )
}